indexer/indexer_tests/editable_map_object_test.cpp

```diff
@@ -451,4 +451,117 @@ UNIT_TEST(EditableMapObject_FromFeatureType)
+    emo.SetTypes(types);
+    TEST_EQUAL(emo.GetLocalizedAllTypes(true), std::string("sport-soccer • leisure-pitch"), ());
+    TEST_EQUAL(emo.GetLocalizedAllTypes(false), std::string("leisure-pitch"), ());
+  }
+
+  {
+    EditableMapObject emo;
+    feature::TypesHolder types;
+    types.Add(classif().GetTypeByPath({"craft", "key_cutter"}));
+    emo.SetTypes(types);
+    TEST_EQUAL(emo.GetLocalizedAllTypes(true), std::string("craft-key_cutter"), ());
+    TEST_EQUAL(emo.GetLocalizedAllTypes(false), std::string(""), ());
+  }
+
+  {
+    EditableMapObject emo;
+    feature::TypesHolder types;
+    types.Add(classif().GetTypeByPath({"amenity", "parking_entrance"}));
+    types.Add(classif().GetTypeByPath({"barrier", "gate"}));
+    emo.SetTypes(types);
+    TEST_EQUAL(emo.GetLocalizedAllTypes(true), std::string("barrier-gate • amenity-parking_entrance"), ());
+    TEST_EQUAL(emo.GetLocalizedAllTypes(false), std::string("amenity-parking_entrance"), ());
+  }
+
+  {
+    EditableMapObject emo;
+    feature::TypesHolder types;
+    types.Add(classif().GetTypeByPath({"barrier", "gate"}));
+    emo.SetTypes(types);
+    TEST_EQUAL(emo.GetLocalizedAllTypes(true), std::string("barrier-gate"), ());
+    TEST_EQUAL(emo.GetLocalizedAllTypes(false), std::string(""), ());
+  }
+
+  {
+    EditableMapObject emo;
+    feature::TypesHolder types;
+    types.Add(classif().GetTypeByPath({"entrance", "main"}));
+    emo.SetTypes(types);
+    TEST_EQUAL(emo.GetLocalizedAllTypes(true), std::string("entrance-main"), ());
+    TEST_EQUAL(emo.GetLocalizedAllTypes(false), std::string(""), ());
+  }
+
+  {
+    EditableMapObject emo;
+    feature::TypesHolder types;
+    types.Add(classif().GetTypeByPath({"entrance", "main"}));
+    types.Add(classif().GetTypeByPath({"barrier", "gate"}));
+    emo.SetTypes(types);
+    TEST_EQUAL(emo.GetLocalizedAllTypes(true), std::string("barrier-gate"), ());
+    TEST_EQUAL(emo.GetLocalizedAllTypes(false), std::string(""), ());
+  }
+
+  {
+    EditableMapObject emo;
+    feature::TypesHolder types;
+    types.Add(classif().GetTypeByPath({"amenity"}));
+    emo.SetTypes(types);
+    TEST_EQUAL(emo.GetLocalizedAllTypes(true), std::string("amenity"), ());
+    TEST_EQUAL(emo.GetLocalizedAllTypes(false), std::string(""), ());
+  }
+
+  {
+    EditableMapObject emo;
+    feature::TypesHolder types;
+    types.Add(classif().GetTypeByPath({"shop"}));
+    emo.SetTypes(types);
+    TEST_EQUAL(emo.GetLocalizedAllTypes(true), std::string("shop"), ());
+    TEST_EQUAL(emo.GetLocalizedAllTypes(false), std::string(""), ());
+  }
+
+  {
+    EditableMapObject emo;
+    feature::TypesHolder types;
+    types.Add(classif().GetTypeByPath({"tourism", "artwork"}));
+    types.Add(classif().GetTypeByPath({"amenity"}));
+    emo.SetTypes(types);
+    TEST_EQUAL(emo.GetLocalizedAllTypes(true), std::string("tourism-artwork"), ());
+    TEST_EQUAL(emo.GetLocalizedAllTypes(false), std::string(""), ());
+  }
```

Comment:
nit: LGTM, but I'd prefer to avoid copy-paste by introducing some helper fixture class here. I can make it after the merge ..